### ISSUE TITLE
Crash at app update

### ISSUE THREAD

**Issue by ovicristurean** [commonsguy/cwac-saferoom#64]
I have an app with no room db, and at update, a db with saferoom in place. What happens when attempting to use the database is this:

2019-08-26 09:51:24.217 31117-31165/ E/AndroidRuntime: FATAL EXCEPTION: pool-5-thread-1
    Process:, PID: 31117
    net.sqlcipher.database.SQLiteException: file is not a database: , while compiling: select count(*) from sqlite_master;
        at net.sqlcipher.database.SQLiteCompiledSql.native_compile(Native Method)
        at net.sqlcipher.database.SQLiteCompiledSql.compile(SQLiteCompiledSql.java:91)
        at net.sqlcipher.database.SQLiteCompiledSql.<init>(SQLiteCompiledSql.java:64)
        at net.sqlcipher.database.SQLiteProgram.<init>(SQLiteProgram.java:89)
        at net.sqlcipher.database.SQLiteQuery.<init>(SQLiteQuery.java:48)
        at net.sqlcipher.database.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:60)
        at net.sqlcipher.database.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:2012)
        at net.sqlcipher.database.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1898)
        at net.sqlcipher.database.SQLiteDatabase.keyDatabase(SQLiteDatabase.java:2647)
        at net.sqlcipher.database.SQLiteDatabase.openDatabaseInternal(SQLiteDatabase.java:2577)
        at net.sqlcipher.database.SQLiteDatabase.openDatabase(SQLiteDatabase.java:1243)
        at net.sqlcipher.database.SQLiteDatabase.openOrCreateDatabase(SQLiteDatabase.java:1318)
        at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:166)
        at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:135)
        at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:131)
        

Do you have any idea about what might be causing this? Thanks

**Comment by hannesa2** [comment]
I struggle with similar issues. Currently it looks like a background task like JobScheduler or similar causes this.
I tested with 
```
class MyApplication : Application() {
   ...
    override fun onTerminate() {
        appDatabase.close()
        super.onTerminate()
    }
```
but then you run into "_DB not open_" errors. When and how to open db again ... ?

Even making a PR in this repo and based SqlCipher repo is a kind of senseless. You have to ask yourself, if you want to be depending on this repo. It has a promising feature, but I at least spent to much time with this not working library

**Comment by commonsguy** [comment]
@hannesa2: `onTerminate()` on an `Application` is not called &mdash; see [the documentation](https://developer.android.com/reference/android/app/Application.html?hl=en#onTerminate()). 

@ovicristurean: That comes from SQLCipher. The most common cause is that the passphrase is invalid.

**Comment by hannesa2** [comment]
Both statements are not true

## A:
> @hannesa2: `onTerminate()` on an `Application` is not called — see [the documentation](https://developer.android.com/reference/android/app/Application.html?hl=en#onTerminate()).
> 
How do yout explain after my close statement this one ?
<img width="666" alt="image" src="https://user-images.githubusercontent.com/3314607/63687706-96afb500-c805-11e9-9fc5-48fc42132549.png">


## B:
> @ovicristurean: That comes from SQLCipher. The most common cause is that the passphrase is invalid.

I use a stable, always the same, password

**Comment by commonsguy** [comment]
> How do yout explain after my close statement this one ?

Since I do not have the source code that corresponds to this stack trace &mdash; let alone a project that reproduces the crash &mdash; I cannot fathom a guess. Give me a way to reproduce the problem, and I will happily look into it.

**Comment by ovicristurean** [comment]
In my case, I think the problem is that I also have GreenDAO ORM which also uses SQLCipher. Perhaps SafeRoom is changing the passphrase, and the secure key which I used for GreenDao cannot be used to open the writable database, since it has been changed by SafeRoom. I noticed that at app update, if I open the GreenDAO db without it being encrypted, everything is fine. Of course, this is not the optimal solution, so I am currently migrating all my GreenDao entities to Room, and then removing the dependency completely. Will let you know if this will be the final solution in my case, because I believe this would be useful to some other people experiencing the same issues.
